cookbooks/bcpc-hadoop/templates/default/hv_hive-env.sh.erb

```diff
@@ -6,3 +6,4 @@ export HIVE_CONF_DIR=/etc/hive/conf.<%= node.chef_environment %>
 # export HIVE_AUX_JARS_PATH=
 export HIVE_AUX_JARS_PATH=/usr/lib/hcatalog/share/hcatalog/hcatalog-core.jar
 
+export HADOOP_HEAPSIZE=<%=node[:bcpc][:hadoop][:heapsize] %>
```

Comment:
Since the heap size is for hadoop and not specifically for hive, why not make it in bcpc-hadoop/templates/default/hdp_hadoop-env.sh.erb ?